src/main/java/org/jnode/fs/xfs/extent/DataExtent.java

```diff
@@ -1,14 +1,24 @@
 package org.jnode.fs.xfs.extent;
 
+import lombok.Getter;
 import org.jnode.fs.xfs.XfsFileSystem;
 import org.jnode.fs.xfs.XfsObject;
 import org.jnode.fs.xfs.superblock.Superblock;
 
 /**
  * A data extent ('xfs_bmbt_rec_t' packed, or 'xfs_bmbt_irec_t' unpacked).
- *
+ * It is the packed format in this class, and the unpacked version is below.
+ * <pre>
+ *   struct xfs_bmbt_irec {
+ *     xfs_fileoff_t br_startoff;
+ *     xfs_fsblock_t br_startblock;
+ *     xfs_filblks_t br_blockcount;
+ *     xfs_exntst_t br_state;
```

Comment:
`state` is declared but not read from the buffer in the constructor. It should also be declared `final`.

Reply:
`initialised` is kind of the `state`. Actually, if you read line75-85, the implements in this class scares me, it is the "packed" one, though the document says it is same to the unpacked one. I may leave it for now.

Reply:
Ah yeah. I had to work with the Wizeline team on this one. https://github.com/Nuix/jnode-fs/pull/5/files/480e4aee1ab6c988e3f5a02d5229d6caad1c0a5c#r772079885 I think what was done originally was incorrect. I had to mess around with bitfields and structures/packed values in my previous job.